lib/encoding.neon

```diff
@@ -0,0 +1,132 @@
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0x80 - 0x8f
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0x90 - 0x9f
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xa0 - 0xaf
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xb0 - 0xbf
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xc0 - 0xcf
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xd0 - 0xdf
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xe0 - 0xef
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0xf0 - 0xff
+]
+
+%FUNCTION base64Encode(src: Bytes): String
+%|
+ |  Function: base64Encode
+ |
+ |  Encodes the provided Bytes into Base64 encoded data.
+ |
+ |  Parameters:
+ |      src - The source data to encode.
+ |
+ |%
+FUNCTION base64Encode(src: Bytes): String
+    VAR encoded: String := ""
+    VAR char1, char2: Number
+    VAR status: State := State.byte1
+
+    LET source: Array<Number> := src.toArray()
+
+    FOR i:= 0 TO src.size()-1 DO
+        LET char: Number := source[i]
+        CASE status
+            WHEN State.byte1 DO
+                encoded.append(Base64EncodeTable[bitwise.shiftRight32(char, 2)])
+                char1 := bitwise.and32(bitwise.shiftLeft32(char, 4), 0b00110000)
+                status := State.byte2
+            WHEN State.byte2 DO
+                encoded.append(Base64EncodeTable[bitwise.or32(char1, bitwise.shiftRight32(char, 4))])
+                char2 := bitwise.and32(bitwise.shiftLeft32(char, 2), 0b00111100)
+                status := State.byte3
+            WHEN State.byte3 DO
+                encoded.append(Base64EncodeTable[bitwise.or32(char2, bitwise.shiftRight32(char, 6))])
+                encoded.append(Base64EncodeTable[bitwise.and32(char, 0b00111111)])
+                status := State.byte1
+        END CASE
+    END FOR
+
+    CASE status
+        % In base64, when the data being encoded is NOT a multiple of 6 bits, needs to be padded.  The following pads the output data
+        % and signals we've hit the end of the encoded data when decoding...
+        WHEN State.byte2 DO
+            encoded.append(Base64EncodeTable[char1])
+            encoded.append("==")
+        WHEN State.byte3 DO
+            encoded.append(Base64EncodeTable[char2])
+            encoded.append("=")
+    END CASE
+    RETURN encoded
+END FUNCTION
+
+%|
+ |  Function: base64Decode
+ |
+ |  Decodes the provided Base64 string data back into binary data.
+ |
+ |  Parameters:
+ |      src - The source Base64 to decode.
+ |
+ |%
+FUNCTION base64Decode(src: String): Bytes
+    VAR char1, char2, char3: Number := 0
+    VAR decoded: Array<Number> := []
+    VAR state: State := State.byte1
+
+    FOR i:= 0 TO src.length() -1 DO
```

Comment:
Spaces around `:=`. Also, be consistent about spaces around `-`. This is different from the loop in the encoder.